specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2022-04-02-preview/privateLinkScopes.json

```diff
@@ -0,0 +1,1095 @@
+      }
+    }
+  },
+  "definitions": {
+    "PrivateLinkScopesResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "TagsResource": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "A container holding only the Tags for a resource, allowing the user to update the tags on a PrivateLinkScope instance."
+    },
+    "KubernetesConfigurationPrivateLinkScope": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Properties that define a Azure Arc PrivateLinkScope resource.",
+          "$ref": "#/definitions/KubernetesConfigurationPrivateLinkScopeProperties"
+        },
+        "systemData": {
+          "readOnly": true,
+          "description": "The system meta data relating to this resource.",
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/PrivateLinkScopesResource"
+        }
+      ],
+      "description": "An Azure Arc PrivateLinkScope definition."
+    },
+    "PrivateLinkScopeValidationDetails": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "publicNetworkAccess": {
+          "description": "Indicates whether machines associated with the private link scope can also use public Azure Arc service endpoints.",
+          "$ref": "#/definitions/PublicNetworkAccessType"
+        },
+        "connectionDetails": {
+          "type": "array",
+          "description": "List of Private Endpoint Connection details.",
+          "items": {
+            "$ref": "#/definitions/ConnectionDetail"
+          }
+        }
+      }
+    },
+    "ConnectionDetail": {
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

I think this id is of the linked resource, not this connection detail object. Could you change it to connectedResourceId or similar? Id implies it's identifying the ConnectionDetail object it belongs to.

Reply:
using common so got fixed